vortex-file/src/read/mod.rs

```diff
@@ -57,20 +59,18 @@ pub type MessageId = Vec<LayoutPartId>;
 /// the message contents.
 #[derive(Debug, Clone)]
 pub struct MessageLocator(pub MessageId, pub ByteRange);
+/// A message that has had its bytes materialized onto the heap.
+#[derive(Debug, Clone)]
+pub struct Message(pub MessageId, pub Bytes);
 
-#[derive(Debug)]
-pub enum BatchRead {
-    ReadMore(Vec<MessageLocator>),
-    Batch(ArrayData),
-}
+pub type BatchRead = MessageRead<ArrayData>;
+pub type SplitRead = MessageRead<RowMask>;
+pub type MetadataRead = MessageRead<Vec<Option<ArrayData>>>;
 
 #[derive(Debug)]
-pub enum MetadataRead {
-    /// Layout has no metadata
-    None,
-    /// Additional IO is required
+pub enum MessageRead<T> {
```

Comment:
Add `None` as a variant and change and remove nested value from Option in all return types